querydsl-core/src/main/java/com/querydsl/core/support/Normalization.java

```diff
@@ -26,15 +26,13 @@
 
     // TODO simplify
     private static final Pattern FULL_OPERATION = Pattern.compile(
-            "(?<![\\d\\*/\"' ])" + "(\\b|\\(|\\s+)"  +
+            "(?<![\\d\\*/\"\\?' ])" + "(\\b|\\(|\\s+)"  +
             "(" + NUMBER + WS + "[+\\-/*]" + WS + ")+" + NUMBER + WS +
             "(?![\\d\\*/\"' ])");
 
     private static final Pattern[] OPERATIONS = {
-            Pattern.compile(NUMBER + WS + "\\*" + WS + NUMBER),
-            Pattern.compile(NUMBER + WS + "/" + WS + NUMBER),
-            Pattern.compile(NUMBER + WS + "\\+" + WS + NUMBER),
-            Pattern.compile(NUMBER + WS + "\\-" + WS + NUMBER)
+            Pattern.compile(NUMBER + WS + "(\\*|/)" + WS + NUMBER),
+            Pattern.compile(NUMBER + WS + "(\\+|\\-)" + WS + NUMBER),
```

Comment:
>in special positions

Yes, you are very right, but what I meant was that escaping isn't necessary in this case, since the '-' is not in a special position.
+1 for you.